src/windows/common/hcs.cpp

```diff
@@ -98,32 +100,39 @@ wsl::windows::common::hcs::unique_hcs_system wsl::windows::common::hcs::CreateCo
 
 std::vector<std::string> wsl::windows::common::hcs::GetProcessorFeatures()
 {
-    ExecutionContext context(Context::HCS);
+    static std::vector<std::string> g_processorFeatures;
+    static std::once_flag flag;
+    std::call_once(flag, [&]() {
+        ExecutionContext context(Context::HCS);
 
-    wil::unique_cotaskmem_string result;
-    THROW_IF_FAILED(::HcsGetServiceProperties(c_processorCapabilitiesQuery, &result));
+        wil::unique_cotaskmem_string result;
+        THROW_IF_FAILED(::HcsGetServiceProperties(c_processorCapabilitiesQuery, &result));
 
-    const auto properties = wsl::shared::FromJson<ServicePropertiesResponse<PropertyResponse<ProcessorCapabilitiesInfo>>>(result.get());
+        const auto properties =
+            wsl::shared::FromJson<ServicePropertiesResponse<PropertyResponse<ProcessorCapabilitiesInfo>>>(result.get());
 
-    const auto& response = properties.PropertyResponses.at(c_processorCapabilities);
-    if (response.Error)
-    {
-        THROW_HR_MSG(static_cast<HRESULT>(response.Error->Error), "%hs", response.Error->ErrorMessage.c_str());
-    }
+        const auto& response = properties.PropertyResponses.at(c_processorCapabilities);
+        if (response.Error)
+        {
+            THROW_HR_MSG(static_cast<HRESULT>(response.Error->Error), "%hs", response.Error->ErrorMessage.c_str());
+        }
+
+        g_processorFeatures = std::move(response.Response.ProcessorFeatures);
```

Comment:
nit: this doesn't actually move the structure since `response` is a const reference